lib/ecto/migration.ex

```diff
@@ -1595,8 +1595,9 @@ defmodule Ecto.Migration do
 
   defp pk_opts_to_tuple(opts) do
     opts = Keyword.put(opts, :primary_key, true)
+    inferred_type = [:start_value, :increment, :cache, :clause] |> Enum.any?(&Keyword.has_key?(opts, &1)) && :bigint || :bigserial
```

Comment:
I would prefer if the type is explicitly declared, instead of inferring. The clearer, the best. Can we please revert this change?

Reply:
@josevalim Sure. Was gonna ask you if you’re ok with switching to identity as default? 

I wonder if these arguments are mutually exclusive. 🧐

Reply:
Doing this would be backwards incompatible as it would change all existing migrations. It is not an option to change it.

Reply:
@josevalim Just to check with you, I know you can pass this as a config to Ecto to set a global override.

Reply:
Yes, exactly, that's the way to make it the default in your apps. Perhaps it is something Phoenix can do in its generators for PG.